rust/private/rustc.bzl

```diff
@@ -1038,9 +1047,6 @@ def construct_arguments(
             data_paths,
         ))
 
-    # Ensure the sysroot is set for the target platform
-    env["SYSROOT"] = toolchain.sysroot
```

Comment:
This should probably be gated behind the new feature flag.

Reply:
Per https://github.com/bazelbuild/rules_rust/issues/2039, `SYSROOT` is no-op. Are you expecting it might be used in user's source code?

Reply:
I also believe that to be true but I don't think it hurts to gate this. Maybe some power users have done something with this flag. This is a change to the core rules so I think we should have a controlled rollout.

Reply:
Should we grate `SYSROOT` and `--sysroot` behind two different feature flags? Even though it sounds redundant, I think it's a good practice since they're pretty orthogonal in this context - one is used by rustc and one is used by user's source code.

Reply:
I personally think we'd be fine without a second flag. I see this change as correcting the definition of `--sysroot` by moving it from the unused `SYSROOT` environment variable to the actual `--sysroot` rustc flag. It would be nice to delete the env var unless it's consumed by rustc in some meaningful way (which I don't believe it is). Thus, I feel the new feature flag should gate both to more clearly communicate the transition. Does that sound reasonable?

Reply:
> I see this change as correcting the definition of --sysroot by moving it from the unused SYSROOT environment variable to the actual --sysroot rustc flag.

Because we have came to the conclusion that `SYSROOT` is not used by `rustc` but might be used by end users (as stated in this thread), `SYSROOT` and `--sysroot` are orthogonal.

Here is why:

Before adding `--sysroot` to `contruct_arguments`, `SYSROOT` is not used anywhere in rust internals but only (possibly) end users. Adding `--sysroot` does not change that fact for `SYSROOT`. Hence, this PR is not meant to move to `--sysroot` from `SYSROOT`. 

Adding `--sysroot` flag to rustc ensures that rustc finds stdlibs and binaries from a directory generated by `rust_toolchain`.

So ideally, we should just keep `SYSROOT` as is in this PR. We can clean up `SYSROOT` in the future or in a separate PR.

Reply:
that also sounds fine to me 😄